test/com/google/javascript/jscomp/GoldenFileComparer.java

```diff
@@ -104,6 +105,33 @@ private static void compileAndCompare(
     compare(compiledSource, referenceSource);
   }
 
+  /**
+   * Compile sourceFileName with given CompilerOptions and check that all lines of code from
+   * goldenFileName are present in compiled sourceFile.
+   */
+  public static void compileAndCompareSubsetOfActualToExpected(
+      String goldenFileName, CompilerOptions options, String sourceFileName) throws Exception {
+    List<SourceFile> sourceFiles = ImmutableList.of(readSource(sourceFileName));
+    List<SourceFile> externsFiles = ImmutableList.of();
+    String compiledSource = compile(externsFiles, sourceFiles, options);
+    String referenceSource = readFile(toFullPath(goldenFileName));
+
+    String[] compiledLines = compiledSource.split("\n");
+    String[] goldenLines = referenceSource.split("\n");
+
+    int expectedNumberOfIdenticalLines = goldenLines.length;
+
+    int identicalLinesBetweenFiles = 0;
+
+    for (int i = 0; i < compiledLines.length; i++) {
+      if (goldenLines[identicalLinesBetweenFiles].equals(compiledLines[i])) {
+        identicalLinesBetweenFiles++;
+      }
+    }
+
+    assertThat(identicalLinesBetweenFiles).isEqualTo(expectedNumberOfIdenticalLines);
```

Comment:
If this assertion fails the error message won't be very helpful.

Are the identical lines always going to be the last `expectedNumberOfIdenticalLines` of `compiledLines`? If so we could instead just truncate `compiledLines` then expect exact string equality.

Reply:
The beginning and end lines of the compiledLines are what is expected to match with expectedNumberOfIdenticalLines. Changed to add a message of what is the first line that is not found.